src/main/scala/devices/debug/Debug.scala

```diff
@@ -476,14 +511,19 @@ class TLDebugModuleOuter(device: Device)(implicit p: Parameters) extends LazyMod
     io.innerCtrl.bits.hartsel      := Mux(DMCONTROLWrEn, DMCONTROLWrData.hartsello, DMCONTROLReg.hartsello)
     io.innerCtrl.bits.resumereq    := DMCONTROLWrEn & DMCONTROLWrData.resumereq    // This bit is W1
     io.innerCtrl.bits.ackhavereset := DMCONTROLWrEn & DMCONTROLWrData.ackhavereset
+    io.innerCtrl.bits.hrmask       := hrmask
     if (supportHartArray) {
       io.innerCtrl.bits.hasel        := Mux(DMCONTROLWrEn, DMCONTROLWrData.hasel, DMCONTROLReg.hasel)
       io.innerCtrl.bits.hamask       := hamask
     }
 
     io.ctrl.ndreset := DMCONTROLReg.ndmreset
     io.ctrl.dmactive := DMCONTROLReg.dmactive
-
+    io.coreResetReq.foreach { req =>
+      for (component <- 0 until nComponents) {
+        req(component) := DMCONTROLReg.hartreset & hartSelected(component)
```

Comment:
I was somewhat confused by the use of `coreReset` when the spec calls this "hart" reset, and we either call them "harts" or "components" in this code. Is there any reason to use coreReset here?